spec/support/shared_tests.rb

```diff
@@ -0,0 +1,75 @@
+include Rails.application.routes.url_helpers
+
+shared_examples_for "an auth controller" do
+  context "when using auth_provider authentication" do
+    context "is successful" do
+      before { request.env["omniauth.auth"] = OmniAuth.config.mock_auth[auth_provider] }
+
+      it "redirects to home page" do
```

Comment:
in some of the tests there is horizontal whitespace in between it statements and in others they are stacked with no whitespace. I personally am a fan of the horizontal whitespace for readability but either way you choose to go as a team you want to be consistent.